core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelector.java

```diff
@@ -64,8 +65,8 @@
     private final boolean includeCoordinator;
     private final AtomicReference<Supplier<NodeMap>> nodeMap;
     private final int minCandidates;
-    private final int maxSplitsPerNode;
-    private final int maxPendingSplitsPerTask;
+    private final long maxSplitsWeightPerNode;
```

Comment:
I wonder what is the memory implication of a split on a worker node. Should there be a limit for maximum pending splits on a node to avoid scheduling to many split objects that can in theory push the node out of memory (in some degenerate case?)